main/src/library/Array.flix

```diff
@@ -871,6 +871,24 @@ namespace Array {
     pub def count(f: a -> Bool, a: Array[a]): Int32 & Impure =
         foldLeft((b,x) -> if (f(x)) b + 1 else b, 0, a) as & Impure
 
+    ///
+    /// Returns the sum of all elements in the array `a`.
+    ///
+    pub def sum(a: Array[Int32]): Int32 & Impure =
+        sumWith(x -> x, a)
```

Comment:
@mlutze @jaschdoc What is more idomatic in Flix? To write `x -> x` or to write `identity`?

Reply:
I have a slight preference for `identity` just because it's more uniform. (`x -> x` could just as well be `z -> z` or `apples -> apples`.)

I guess there's also a tiny perk that you don't have to think about if you already have `x` in scope, and if you make changes and add an `x` somewhere, you won't have to deal with a shadowing error.

Not exactly an overwhelming argument though.

Reply:
Good enough for me, let us favor identity